src/Microsoft.IdentityModel.Tokens/Exceptions/ValidationFailure.cs

```diff
@@ -0,0 +1,48 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// The reason for being unable to validate
+    /// </summary>
+    public enum ValidationFailure
+    {
+        /// <summary>
+        /// Indicates no validation failures
+        /// </summary>
+        None = 0b_0000_0000_0000_0000, // 0
+        /// <summary>
+        /// Indicates that the lifetime was invalid
+        /// </summary>
+        InvalidLifetime = 0b_0000_0000_0000_0001, // 1
+        /// <summary>
+        /// Indicates that the issuer was invalid
+        /// </summary>
+        InvalidIssuer = 0b_0000_0000_0000_0010, // 2
+    }
```

Comment:
I might be misunderstanding something.... do we need an enum that represents when both are invalid?

Reply:
The idea of using a bitflag: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum#enumeration-types-as-bit-flags

That way we can represent a combination. And then the end user will do .HasFlag() to determine if a particular value is set.

If you think that is confusing, we can use distinct enums values